diff --git a/contrib/zlib/CMakeLists.txt b/contrib/zlib/CMakeLists.txt
index 0e8a4b1e..c718fcf1 100644
--- a/contrib/zlib/CMakeLists.txt
+++ b/contrib/zlib/CMakeLists.txt
@@ -198,11 +198,9 @@ endif(MINGW)
 
 add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
 
-IF(WIN32)
-    INSTALL( TARGETS zlibstatic
-             EXPORT "${TARGETS_EXPORT_NAME}"
-             LIBRARY DESTINATION ${ASSIMP_LIB_INSTALL_DIR}
-             ARCHIVE DESTINATION ${ASSIMP_LIB_INSTALL_DIR}
-             RUNTIME DESTINATION ${ASSIMP_BIN_INSTALL_DIR}
-             COMPONENT ${LIBASSIMP_COMPONENT})
-ENDIF()
+INSTALL( TARGETS zlibstatic
+            EXPORT "${TARGETS_EXPORT_NAME}"
+            LIBRARY DESTINATION ${ASSIMP_LIB_INSTALL_DIR}
+            ARCHIVE DESTINATION ${ASSIMP_LIB_INSTALL_DIR}
+            RUNTIME DESTINATION ${ASSIMP_BIN_INSTALL_DIR}
+            COMPONENT ${LIBASSIMP_COMPONENT})
