diff --git a/src/command.c b/src/command.c
index bab5ca3e..731a6d12 100644
--- a/src/command.c
+++ b/src/command.c
@@ -133,6 +133,7 @@ static SF_FORMAT_INFO const major_formats [] =
 	{	SF_FORMAT_MAT4,		"MAT4 (GNU Octave 2.0 / Matlab 4.2)",	"mat"	},
 	{	SF_FORMAT_MAT5,		"MAT5 (GNU Octave 2.1 / Matlab 5.0)",	"mat"	},
 	{	SF_FORMAT_MPC2K,	"MPC (Akai MPC 2k)",					"mpc"	},
+	{	SF_FORMAT_MP3,		"MP3 (So-called MP3 file)",				"mp3"	},
 #if HAVE_EXTERNAL_XIPH_LIBS
 	{	SF_FORMAT_OGG,		"OGG (OGG Container format)",			"ogg"	},
 #endif
