diff --git a/src/command.c b/src/command.c
index bab5ca3e..1e6c93f8 100644
--- a/src/command.c
+++ b/src/command.c
@@ -134,7 +134,7 @@ static SF_FORMAT_INFO const major_formats [] =
 	{	SF_FORMAT_MAT5,		"MAT5 (GNU Octave 2.1 / Matlab 5.0)",	"mat"	},
 	{	SF_FORMAT_MPC2K,	"MPC (Akai MPC 2k)",					"mpc"	},
 #if HAVE_EXTERNAL_XIPH_LIBS
-	{	SF_FORMAT_OGG,		"OGG (OGG Container format)",			"oga"	},
+	{	SF_FORMAT_OGG,		"OGG (OGG Container format)",			"ogg"	},
 #endif
 	{	SF_FORMAT_PAF,		"PAF (Ensoniq PARIS)", 					"paf"	},
 	{	SF_FORMAT_PVF,		"PVF (Portable Voice Format)",			"pvf"	},
