diff --git a/qtwebengine/src/3rdparty/chromium/third_party/webrtc/webrtc.gni b/qtwebengine/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
index 56a1b0deca0..c07b63983f8 100644
--- a/qtwebengine/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
+++ b/qtwebengine/src/3rdparty/chromium/third_party/webrtc/webrtc.gni
@@ -150,7 +150,7 @@ declare_args() {
   # Enabling H264 when building with MSVC is currently not supported, see
   # bugs.webrtc.org/9213#c13 for more info.
   rtc_use_h264 =
-      proprietary_codecs && !is_android && !is_ios && !(is_win && !is_clang)
+      proprietary_codecs && !is_android && !is_ios
 
   # By default, use normal platform audio support or dummy audio, but don't
   # use file-based audio playout and record.
